__tests__/components/TransactionList/TransactionRow.spec.ts

```diff
@@ -0,0 +1,888 @@
+            UInt64.fromUint(100),
+            mockSignature,
+            PublicAccount.createFromPublicKey(currentSigner.publicKey, NetworkType.TEST_NET),
+        );
+
+        const signedTransaction = currentSigner.sign(
+            aggregateTransaction,
+            '57F7DA205008026C776CB6AED843393F04CD458E0AA2D9F1D5F31A402072B2D6',
+        );
+
+        const hashLockTransaction = HashLockTransaction.create(
+            mockDeadline,
+            new Mosaic(new MosaicId(networkMock.currency.mosaicIdHex), UInt64.fromUint(10)),
+            UInt64.fromUint(10),
+            signedTransaction,
+            NetworkType.TEST_NET,
+            UInt64.fromUint(100),
+            mockSignature,
+            PublicAccount.createFromPublicKey(currentSigner.publicKey, NetworkType.TEST_NET),
+        );
+
+        const preparedData = [
+            {
+                transactionType: 'other transaction deadline in 6 hours',
+                mockTransaction: mockTransferTransaction(),
+                expectedDate: '2019-11-10 22:00:00',
+            },
+            {
+                transactionType: 'Aggregate Bond transaction deadline in 48 hours',
+                mockTransaction: mockAggregateTransaction(),
+                expectedDate: '2019-11-09 00:00:00',
+            },
+            {
+                transactionType: 'Hash Lock transaction deadline in 6 hours',
+                mockTransaction: hashLockTransaction,
+                expectedDate: '2019-11-10 18:00:00',
+            },
+        ];
+
+        preparedData.map((test) => runTransactionDateTests(test));
+    });
+
+    describe('isOptinPayoutTransaction', () => {
+        test('returns false when transaction is not exist', async () => {
+            // Arrange:
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile, currentAccount: WalletsModel1 },
+                { transaction: mockTransferTransaction() },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            vm.transaction = undefined;
+
+            // Act + Assert:
+            expect(vm.isOptinPayoutTransaction).toBe(false);
+        });
+
+        const runIsOptinPayoutTransactionTests = (state) => {
+            // Arrange:
+            const wrapper = getTransactionRowWrapper(state, { transaction: mockAggregateTransaction() });
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act + Assert:
+            expect(vm.isOptinPayoutTransaction).toBe(true);
+        };
+
+        test('returns true when transaction announced by NGL (testnet)', async () => {
+            runIsOptinPayoutTransactionTests({ currentProfile: testnetProfile, currentAccount: WalletsModel1 });
+        });
+
+        test('returns true when transaction announced by NGL (mainnet)', async () => {
+            runIsOptinPayoutTransactionTests({ currentProfile: getTestProfile('profile_mainnet'), currentAccount: WalletsModel2 });
+        });
```

Comment:
NGL? 😅

Reply:
https://github.com/symbol/desktop-wallet/blob/fb54c3682e44bc1e022a17f0a34e32af6ebba066/src/components/TransactionList/TransactionRow/TransactionRowTs.ts#L145

I was refer from here 😓

Reply:
we should probably fix that too